File: Python/7py 2/is_ruby_coming.py
"""
|--------------------------------------------------------------------------
| Coding Meetup #3 - Higher-Order Functions Series - Is Ruby coming?
|--------------------------------------------------------------------------
|
"""

def is_ruby_coming(lst): 
   return len(list(filter(lambda i: i['language'] == 'Ruby', lst))) > 0


def is_ruby_coming(lst): 
   return any(x["language"] == "Ruby" for x in lst)


def is_ruby_coming(lst): 
   for i in lst:
      if i['language'] == 'Ruby':
         return True
   return False


def is_ruby_coming(lst): 
   return next((True for i in lst if i['language'] == 'Ruby'), False)

File: PHP/7php 1/is_ruby_coming.php
<?php
/*
|--------------------------------------------------------------------------
| Coding Meetup #3 - Higher-Order Functions Series - Is Ruby coming?
|--------------------------------------------------------------------------
|
*/

function is_ruby_coming($a) {
   return count(array_filter($a, fn($i) => $i['language'] == 'Ruby')) > 0;
}


function is_ruby_coming($a) {
   return in_array('Ruby',array_column($a,'language'));
}


function is_ruby_coming($a) {
   foreach($a as $dev){
      if($dev['language'] == 'Ruby')
      return true;
   }
   return false;
}


function is_ruby_coming($a) {
   $rubyDev = array_reduce($a, function($count, $item) {
      return $item["language"] == "Ruby" ? ++$count : $count; 
   }, 0);
   return $rubyDev > 0 ? true : false;
}